docs/user_deletion.md

```diff
@@ -0,0 +1,18 @@
+# User Deletion
```

Comment:
so we need to highlight that there are three things that can happen or we have built for: 

we need to mention that they should choose the option based on their terms and conditions and privacy policy of the app that they are building. 

1. just soft delete the user (default if no rake task for annon is setup) - no data is deleted
2. soft delete and after a period of time annon the data of the user so there is no PII (personal identifiable information). say how they can set this to be 0 so it happens immediately or how to customise the time duration. 
3. just delete the user permanently immediately when the user requests and how they would do this.